### DISCUSSION TITLE
how-to-install-older-versions-of-homebrew-packages/

### DISCUSSION THREAD

**Discussion by giscus** [nelsonfigueroa/nelson.cloud#18] [Announcements]
# how-to-install-older-versions-of-homebrew-packages/

How to install a specific version of Homebrew packages.

https://nelson.cloud/how-to-install-older-versions-of-homebrew-packages/

**Comment by kubrdom** [comment]
Perfect tutorial, Thanks!

**Reply by nelsonfigueroa** [reply]
Glad you found it useful :)

**Comment by tus512** [comment]
I have to login to say "thank you" to you. Such a great tutorial.

**Comment by ajxb** [comment]
This tutorial is brilliant, clear and concise, thanks for taking the time to write it.

I accidentally upgraded vagrant to the latest version and turns out that it isn't compatible with a critical plugin I use, this tutorial helped me quickly rollback to the previous version and get my system working again without too much fuss.

Saved me many hours of stress!

**Comment by manroygood** [comment]
This got me closer than any other tutorial so far. I found the version I wanted and created a file `hugo.rb`. I then ran `brew intall hugo.rb` and got the error message

```
Error: Homebrew requires formulae to be in a tap, rejecting:
  hugo.rb (/Users/manroygood/Documents/GitHub/manroygood/hugo.rb)

To create a tap, run e.g.
  brew tap-new <user|org>/<repository>
To create a formula in a tap run e.g.
  brew create <url> --tap=<user|org>/<repository>
```

Following another tutorial on this topic, I got as far as creating the new tap with the command `brew tap-new $USER/local-hugo` I think I'm one step away from getting this working. Not quite sure how to follow the last step.

**Reply by nelsonfigueroa** [reply]
Looks like `brew` was [updated](https://github.com/Homebrew/brew/pull/20414) after I wrote this post to prevent installing formulae without setting an env variable.

I installed a random older version of Hugo (v0.146.3) by setting the `HOMEBREW_DEVELOPER` env variable to `true` like so:
```
HOMEBREW_DEVELOPER=true brew install hugo.rb
```

I'll update my post accordingly. Thanks for calling this out!

**Reply by manroygood** [reply]
Thanks! Getting closer, but I'm still not quite there. I was able to follow the line `create a tap` in the error message above, but I'm not sure what to put for `<url>` in the `create a formula` step. 

Do I need to rerun the `curl` command and set the destination to the directory created in the `brew tap-new` step from your tutorial instead of running `brew create`?

**Reply by nelsonfigueroa** [reply]
You should be able to get past the `brew tap-new` part altogether if `HOMEBREW_DEVELOPER=true` set.

In my case, after downloading the `.rb` file (via `curl` or otherwise) I was able to run `HOMEBREW_DEVELOPER=true brew install hugo.rb` to install Hugo without getting prompted about a tap or creating a formula.

**Reply by manroygood** [reply]
That works! Thank you so much.
